### PR TITLE
feat: add openapi-mode config

### LINKED ISSUE #1449: OpenAPI output that doesn't consider role privileges
Basically, users would like to get all the schema info through OpenAPI without caring for privileges. This was also discussed in https://github.com/PostgREST/postgrest/issues/1060 and we get this question on gitter chat from time to time. The workaround is cumbersome — adding a hardcoded high privileged JWT for the root endpoint. 

I propose we include a config option called `openapi-follows-acl`, default to true to avoid a breaking change. When this is false we'll use our schema cache info without considering privileges.

### PR DESCRIPTION
**Pull request by steve-chavez**
Fixes #1449. Adds the `openapi-mode` config:

* `openapi-mode="follow-acl"`(default): OpenAPI output follows access control for the JWT role.
* `openapi-mode="ignore-acl"`: OpenAPI output  ignores access control for the JWT role.
* `openapi-mode="disabled"`: disables OpenAPI output, the root endpoint replies with 404 Not Found.

#### Pending

- [ ] Add tests

### CODE REVIEW

File: src/PostgREST/Config.hs [hunk 1]
```diff
@@ -101,6 +103,9 @@ instance Show LogLevel where
   show LogWarn  = "warn"
   show LogInfo  = "info"
 
+data OpenAPIMode = OAIFollowACL | OAIIgnoreACL | OAIDisabled
+  deriving Eq
```
Comment (laurenceisla) [inline src/PostgREST/Config.hs:107]: For "OpenAPI", I like OA better as an acronym than OAI, e.g. `OAFollowACL`, `OADisabled`. WDYT?

File: src/PostgREST/Config.hs [hunk 2]
```diff
@@ -128,6 +133,7 @@ toText conf =
       ,("jwt-secret-is-base64",          T.toLower . show . configJwtSecretIsBase64)
       ,("log-level",                 q . show . configLogLevel)
       ,("openapi-server-proxy-uri",  q . fromMaybe mempty . configOpenApiServerProxyUri)
+      ,("openapi-server-proxy-uri",  q . fromMaybe mempty . configOpenApiServerProxyUri)
```
Comment (laurenceisla) [inline src/PostgREST/Config.hs:136]:
```suggestion
```
Looks like this setting is duplicated.

File: src/PostgREST/Config.hs (hunk 1 above)
Comment (steve-chavez) [reply src/PostgREST/Config.hs:107]: Yeah, that's better. The OAI term from https://github.com/OAI, somehow stuck for me, but it's a different thing.

File: src/PostgREST/Config.hs (hunk 2 above)
Comment (steve-chavez) [reply src/PostgREST/Config.hs:136]: Right, that was a mistake.

**Pull request merged**